sdk/storage/Azure.Storage.DataMovement/src/Shared/LocalTransferCheckpointer.cs

```diff
@@ -125,41 +126,63 @@ public LocalTransferCheckpointer(string folderPath)
             CancellationToken cancellationToken = default)
         {
             CancellationHelper.ThrowIfCancellationRequested(cancellationToken);
-            if (_transferStates.TryGetValue(transferId, out var result))
+            if (_transferStates.TryGetValue(transferId, out JobPlanFile result))
             {
-                return Task.FromResult<int>(result.JobParts.Count);
+                return Task.FromResult(result.JobParts.Count);
             }
             throw Errors.MissingTransferIdCheckpointer(transferId);
         }
 
         /// <inheritdoc/>
-        public override async Task<Stream> ReadableStreamAsync(
+        public override async Task<Stream> ReadJobPlanFileAsync(
+            string transferId,
+            int offset,
+            int length,
+            CancellationToken cancellationToken = default)
+        {
+            Stream copiedStream = new MemoryStream(length);
+
+            CancellationHelper.ThrowIfCancellationRequested(cancellationToken);
+            if (_transferStates.TryGetValue(transferId, out JobPlanFile jobPlanFile))
+            {
+                using (MemoryMappedFile mmf = MemoryMappedFile.CreateFromFile(jobPlanFile.FilePath))
+                using (MemoryMappedViewStream mmfStream = mmf.CreateViewStream(offset, length, MemoryMappedFileAccess.Read))
+                {
+                    await mmfStream.CopyToAsync(copiedStream).ConfigureAwait(false);
+                    copiedStream.Position = 0;
+                    return copiedStream;
+                }
+            }
+            else
+            {
+                throw Errors.MissingTransferIdCheckpointer(transferId);
+            }
+        }
+
+        /// <inheritdoc/>
+        public override async Task<Stream> ReadJobPartPlanFileAsync(
             string transferId,
             int partNumber,
-            long offset,
-            long readSize,
+            int offset,
+            int length,
             CancellationToken cancellationToken = default)
         {
             if (_transferStates.TryGetValue(transferId, out JobPlanFile jobPlanFile))
             {
                 if (jobPlanFile.JobParts.TryGetValue(partNumber, out JobPartPlanFile jobPartPlanFile))
                 {
-                    Stream copiedStream = new MemoryStream(DataMovementConstants.JobPartPlanFile.JobPartHeaderSizeInBytes);
+                    Stream copiedStream = new MemoryStream(length);
```

Comment:
I removed reliance on `JobPartHeaderSizeInBytes` here since after the schema update, we won't know the full length due to variable length. We will take a L here if `length=0` but it's just the capacity so not too big of a deal.